### PR TITLE
Continue cleaning GenTreeStmt.

### PR DESCRIPTION
**Pull request by sandreenko**
This is one of the final set of changes. Contributes to #25899.
No asm diffs, no PIN diffs.

In this PR:
- Rename `GenTreeStmt` to `Statement`;
- Add iterators for statements;
- Return void from GenTreeStmt* fgInsertStmtSomewhere.

Fix comments/headers along the way.

I have a PR that has more upcoming changes #26610, but it is too big to be reviewed/merged, so I split it.

As usual, I recommend to review by commits:
c2f67b7c54: Rename `GenTreeStmt` to `Statement`.                                     
                                                                                     
096ef85f59: Move `Statements::IsPhiDefnStmt` implementation to the header.           
                                                                                     
3f04470c3f: Add iterators for statements.                                            
                                                                                     
So we have fewer places where we access Statement's fields.                           
                                                                                     
1281108bd1: Add a header for `fgInsertStmtListAfter`.
                                                                                     
46d67ffd7c: Change `fgInsertStmtAtBeg` not to return a value.                        
                                                                                     
and add headers for `fgInsertStmtAtBeg` and `fgNewStmtAtBeg`.                        
                                                                                     
720d135b1b: Change `fgInsertStmtBefore` and `fgInsertStmtAfter` not to return.       
                                                                                     
56d5db1403: Change `fgInsertStmtNearEnd` not to return.                              
                                                                                     
3bdd7bb727: Change `fgInsertStmtAtEnd` not to return.

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @CarolEidt @dotnet/jit-contrib

File: src/jit/compiler.h [hunk 1]
```diff
@@ -5254,29 +5254,27 @@ class Compiler
...
+    void fgInsertStmtAtBeg(BasicBlock* block, Statement* stmt);
+    Statement* fgNewStmtAtBeg(BasicBlock* block, GenTree* tree);
 
-    GenTreeStmt* fgInsertStmtAfter(BasicBlock* block, GenTreeStmt* insertionPoint, GenTreeStmt* stmt);
+    void fgInsertStmtAfter(BasicBlock* block, Statement* insertionPoint, Statement* stmt);
 
-public: // Used by linear scan register allocation
-    GenTreeStmt* fgInsertStmtBefore(BasicBlock* block, GenTreeStmt* insertionPoint, GenTreeStmt* stmt);
+public: // Used by objectalloc.
+    void fgInsertStmtBefore(BasicBlock* block, Statement* insertionPoint, Statement* stmt);
```
Comment (erozenfeld) [inline src/jit/compiler.h]: I don't think this comment is useful, I would remove it.

File: src/jit/compiler.h [hunk 2]
```diff
@@ -5254,29 +5254,27 @@ class Compiler
...
-    static bool fgBlockContainsStatementBounded(BasicBlock*  block,
-                                                GenTreeStmt* stmt,
-                                                bool         answerOnBoundExceeded = true);
+    static bool fgBlockContainsStatementBounded(BasicBlock* block, Statement* stmt, bool answerOnBoundExceeded = true);
 #endif
 
-public:
-    GenTreeStmt* fgInsertStmtAtEnd(BasicBlock* block, GenTreeStmt* stmt);
-    GenTreeStmt* fgNewStmtAtEnd(BasicBlock* block, GenTree* tree);
+public: // Used by indirectcalltransformer.
```
Comment (erozenfeld) [inline src/jit/compiler.h]: I don't think this comment is useful, I would remove it.

File: src/jit/flowgraph.cpp [hunk 3]
```diff
@@ -535,86 +535,96 @@ bool Compiler::fgBlockContainsStatementBounded(BasicBlock*  block,
...
-GenTreeStmt* Compiler::fgInsertStmtAtEnd(BasicBlock* block, GenTreeStmt* stmt)
+//------------------------------------------------------------------------
+// fgInsertStmtAtEnd: Insert the given statement at the end of the given basic block.
+//
+// Arguments:
+//   block - the block into which 'stmt' will be inserted;
+//   stmt  - the statement to be inserted.
+//
+// Note:
+//   If the block can be a conditional block, use fgNewStmtNearEnd.
```
Comment (erozenfeld) [inline src/jit/flowgraph.cpp]: Should it be "use fgInsertStmtNearEnd"?

Comment (sandreenko) [reply src/jit/flowgraph.cpp]: Fixed, thanks.

File: src/jit/gentree.cpp [hunk 4]
```diff
@@ -11468,24 +11468,11 @@ void Compiler::gtDispArgList(GenTreeCall* call, IndentStack* indentStack)
...
-//    indentStack  - the specification for the current level of indentation & arcs
-//
-// Return Value:
-//    None.
-//
-// Assumptions:
-//    'tree' must be a GT_LIST node
-
-void Compiler::gtDispStmtList(GenTreeStmt* stmts, IndentStack* indentStack /* = nullptr */)
+void Compiler::gtDispBlockStmts(BasicBlock* block)
```
Comment (CarolEidt) [inline src/jit/gentree.cpp]: This needs a function header

**Pull request merged**